### ISSUE TITLE
e2e tests are broken

### ISSUE THREAD

**Issue by fejta** [bazelbuild/rules_k8s#642]
An example failure: https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/bazelbuild_rules_k8s/638/pull-rules-k8s-e2e/1366496761295671296

Problem is that no longer has the ability to create the externally-accessible load balancers that the e2e tests currently depend on, making this part of the e2e tests fail.

Tests create an external LB: 
https://github.com/bazelbuild/rules_k8s/blob/f1c6399cdd691b7aca90073398e8f690ec8992c6/examples/hellohttp/e2e-test.sh#L45-L48

When then wait for an IP address:https://github.com/bazelbuild/rules_k8s/blob/f1c6399cdd691b7aca90073398e8f690ec8992c6/examples/hellohttp/e2e-test.sh#L51-L61

But we will never get one:
https://github.com/bazelbuild/rules_k8s/blob/f1c6399cdd691b7aca90073398e8f690ec8992c6/examples/util.sh#L60-L70

Basically we need to either fix the project to regain this ability or else update the tests to no longer require it.

For the latter situation, I would expect to be able to use `kubectl port-forward` to provide a localhost IP address that connects to the service on the cluster, probably with minor changes to the [https://github.com/bazelbuild/rules_k8s/blob/f1c6399cdd691b7aca90073398e8f690ec8992c6/examples/hellohttp/service.yaml](service).

**Comment by f0rmiga** [comment]
I'd starting by asking _why are we poking the services at all_?
If we trust the k8s API (on a correctly configured cluster), I don't see a need to exercise the examples business logic.
We can make assertions based on the k8s events.

If I'm missing something and _we **must** exercise the business logic_, then `kubectl port-forward` indeed sounds a good candidate.

**Comment by chrislovecnm** [comment]
@f0rmiga can we change the CI to not create services? I would agree that we do not need to do this because we are essentially testing the k8s API and not rules_k8s.  We need to validate that we are communicating the API correctly and that we are installing valid k8s objects, but not need to create services.

**Comment by f0rmiga** [comment]
@chrislovecnm I don't think _creating_ the service is the problem. We can change the assertion from _poking the service_ to _asserting all the resources were created as expected_.

Working on it now.

**Comment by chrislovecnm** [comment]
> Problem is that no longer has the ability to create the externally-accessible load balancers that the e2e tests currently depend on, making this part of the e2e tests fail.

@f0rmiga the way I am reading @fejta's comment is that we cannot create ELB's.  We can create an internal service.

**Comment by fejta** [comment]
Correct, we can create a kubernetes resource that requests an external IP, and this GCP project has been configured to disallow this.

An internal service should not have this problem, so long as we can figure out a way to route packets from the test pod to this service.

Alternatively abandoning the e2e suite and just validating the data we will send kubectl is another strategy. It will definitely be faster but probably will cause us to lose the current integration coverage that rules_k8s will deploy local changes.

AKA right now I can edit //some/webserver/something.go and then calling bazel run //some/websever:dev.apply will build this server, put it into and image and kubectl replace the webserver deployment with this new image that includes my local changes.

Can probably validate this by just looking at the tarball hashes, but either way it is an improvement of the current state of affairs of everything failing tests.

**Comment by chases2** [comment]
#650 removes the broken e2e tests, but fixing and reimplementing these tests still needs to be done.

The approaches that fejta metioned sound good to me. I think validating the configuration might be best, but kubectl port-forward sounds good too.
